dir-list-webpack-plugin.js

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import fs from "fs";
+
+export default class DirListWebpackPlugin {
+  constructor(options) {
+    this.options = {
+      directory: undefined,
+      filename: undefined,
+      ...options,
+    };
+  }
+
+  apply(compiler) {
+    compiler.plugin("emit", (compilation, callback) => {
+      if (!this.options.directory) {
+        compilation.errors.push("DirListWebpackPlugin directory undefined");
+        callback();
+        return;
+      }
+      if (!this.options.filename) {
+        compilation.errors.push("DirListWebpackPlugin filename undefined");
+        callback();
+        return;
+      }
+
+      fs.readdir(this.options.directory, (err, files) => {
+        if (err) {
+          compilation.errors.push(`DirListWebpackPlugin couldn't read ` +
+                                  `directory $(this.options.directory)`);
+          callback();
+          return;
+        }
+
+        // Ignore dotfiles.
+        const output = JSON.stringify(files.filter((x) => x.charAt(0) !== ".");
```

Comment:
ESLint really doesn't like this line. I think you're missing a closing `)` before the `;`.